opengrok-indexer/src/main/java/org/opengrok/indexer/history/RepositoryFactory.java

```diff
@@ -62,20 +68,43 @@
         new SSCMRepository()
     };
 
+    private static final Map<String, Class<? extends Repository>> byName = new HashMap<>();
+
+    static {
+        final String REPOSITORY = "Repository";
```

Comment:
hopefully everyone will name respository classes with this suffix. Assert this in a test ?

Reply:
OK added `testRepositoryFactoryEveryImplIsNamedAsRepository`